single-node.sh

```diff
@@ -114,223 +114,67 @@ build)
-            env PYTHONPATH="${dashboard_python_path}" "${TARGET}"/dashboard-env/bin/python manage.py loaddata "${TARGET}"/dashboard/fixtures/targetdevice.json
-        popd
-
-        switch_state_to node
-
-        ;&
-    node)
-        if [ -z "$(which node)" ]; then
-            info "Installing Node.js"
-            node=node-v"${NODE_VER}"-linux-x64.tar.xz
-            sudo -u "${USER}" curl -o "${TARGET}/${node}" https://nodejs.org/dist/v"${NODE_VER}/${node}"
-            sudo -u "${USER}" tar xJvf "${TARGET}/${node}" -C "${TARGET}"
-            tar=${node%.*}
-            dir=${tar%.*}
-            sudo -u "${USER}" mv "${TARGET}"/"${dir}" "${TARGET}"/node
-            sudo -u "${USER}" rm "${TARGET}"/"${node}"
-        else
-            info "Node.js is already installed"
-        fi
-
-        switch_state_to chroots
-
-        ;&
-    chroots)
-        info "creating chroots environments"
+    if [ -z "$2" ]; then
+        recipe=fast
 
         chroots=(
-            "devuan-jessie-armhf,rpi-3-b"
-            "raspbian-stretch-armhf,rpi-3-b"
-            "ubuntu-xenial-armhf,rpi-2-b"
-            "ubuntu-bionic-arm64,rpi-3-b"
-            "ubuntu-bionic-armhf,rpi-3-b"
-        )
-
-        pushd "${TARGET}"/pieman
-            for chroot in "${chroots[@]}"; do
-                IFS=',' read -r -a pieces <<< "${chroot}"
-
-                env CREATE_ONLY_CHROOT=true \
-                    OS="${pieces[0]}" \
-                    PROJECT_NAME="${pieces[0]}" \
-                    DEVICE="${pieces[1]}" \
-                    PYTHON="${TARGET}"/pieman-env/bin/python \
-                ./pieman.sh
-
-                mv build/"${pieces[0]}"/chroot "${TARGET}/${pieces[0]}"
-            done
-        popd
-
-        switch_state_to success
-
-        ;&
-    success)
-        sudo -u "${USER}" sh -c "echo ${TARGET} > cusdeb"
-
-        switch_state_to init
+                "devuan-jessie-armhf"
+                "raspbian-stretch-armhf"
+                "ubuntu-xenial-armhf"
+                "ubuntu-bionic-arm64"
+                "ubuntu-bionic-armhf"
+            )
+
+        for chroot in "${chroots[@]}"; do
+            lines="$(find "$TARGET" -maxdepth 1 -name "$chroot"| wc -l)"
+            if [ "$lines" -eq 0  ]; then
+                fatal "target directory do not have nessesary build - $chroot, so use full rebuild."
+                exit 1
+            fi
+        done
+    else
+        recipe="$2"
+    fi
 
-        stop_containers
+    resume_state
+    state="$(get_state)"
 
-        success "development environment is done"
+    if [ "${state}" = "init" ] || [ "${state}" = "clone" ]; then
+        info "cleaning up"
+
+        case $recipe in
+        fast)
+            pushd "$TARGET"
+                dirs_to_delete="$(find . -maxdepth 1 ! -name '*arm64' ! -name '*armhf' ! -name '*.')"
```

Comment:
Let's call the variable `to_be_deleted` or something. The point is that there're not only directories in the target directory.